easybuild/tools/filetools.py

```diff
@@ -2435,8 +2435,16 @@ def copy_file(path, target_path, force_in_dry_run=False):
             else:
                 mkdir(os.path.dirname(target_path), parents=True)
                 if path_exists:
-                    shutil.copy2(path, target_path)
-                    _log.info("%s copied to %s", path, target_path)
+                    try:
+                        # on at least some systems, copying read-only files with shutil.copy2() spits a PermissionError
+                        # but the copy still succeeds
+                        shutil.copy2(path, target_path)
+                        _log.info("%s copied to %s", path, target_path)
+                    except PermissionError as err:
+                        # double-check whether the copy actually succeeded
+                        if not os.path.exists(target_path):
```

Comment:
Hmm, it's not really sufficient if it exists though, contents could be different?

Reply:
good point, I've amended that to: `if not os.path.exists(target_path) or not os.path.samefile(path, target_path):`

Reply:
Is that the right option? That's still not checking the contents I don't believe, for that I thought you need `filecmp.cmp(a, b, shallow=False)` (after soke homework in my other PR)

Reply:
ugh, you're right
I'll fix that once the python2 framework test has finished, so I can see if that at least works